src/data/nat/factorial.lean

```diff
@@ -159,6 +159,28 @@ begin
   exact add_factorial_succ_le_factorial_add_succ i h,
 end
 
+lemma pow_sub_mul_factorial_le_factorial {n m : ℕ} (hnm : n ≤ m) : n ^ (m - n) * n! ≤ m! :=
+begin
+  induction m,
+  have hn : n = 0, exact le_zero_iff.mp hnm,
+  rw hn,
+  simp only [le_refl, mul_one, nat.nat_zero_eq_zero, nat.factorial_zero, nat.sub_self, pow_zero],
+  by_cases n ≤ m_n,
+  rw [succ_sub, nat.factorial_succ, pow_succ, mul_assoc],
+  apply mul_le_mul hnm (m_ih h),
+  apply zero_le,
+  apply zero_le,
+  exact h,
+  rw not_le at h,
+  rw factorial_succ,
+  have hn : m_n.succ ≤ n, exact succ_le_iff.mpr h,
+  have hn' : n = m_n.succ, exact le_antisymm hnm hn,
+  rw <- hn',
+  rw [nat.sub_self, pow_zero, one_mul, hn'],
+  simp only [le_refl, mul_le_mul_left, nat.factorial_succ],
+end
```

Comment:
```suggestion
lemma pow_sub_mul_factorial_le_factorial {n m : ℕ} (hnm : n ≤ m) : n! * n ^ (m - n) ≤ m! :=
begin
  suffices : n! * (n + 1) ^ (m - n) ≤ m!,
  { have := pow_le_pow_of_le_left (nat.zero_le _) (le_succ n) (m - n),
    sorry },
  convert nat.factorial_mul_pow_le_factorial,
  exact (nat.add_sub_of_le hnm).symm
end
```

I'm too lazy to fill the sorry, but no need to reinvent the wheel (also note the changed argument order, I think this is better, but it's just one `mul_comm` somewhere anyways)